Formula/tinker.rb

```diff
@@ -59,7 +59,7 @@ def ruby_version
   #
   # @return [Object] Reference to metadata object loaded from a gem.
   def metadata
-    @metadata ||= YAML.load(`tar -xOf #{tinker_gem_path} metadata.gz | gzip -dc`)
+    @metadata ||= YAML.load(`tar -xOf #{tinker_gem_path} metadata.gz | gzip -dc`, permitted_classes: [Gem::Specification, Gem::Version, Gem::Dependency, Gem::Requirement, Gem::Platform, Time, Symbol])
```

Comment:
These were used as a reference for this solution.  
[Upgrading to Ruby 3.1 causes Psych::DisallowedClass exception when using YAML.load_file](https://stackoverflow.com/questions/71332602/upgrading-to-ruby-3-1-causes-psychdisallowedclass-exception-when-using-yaml-lo) 
https://www.rubydoc.info/stdlib/psych/Psych.safe_load
![Screen Shot 2024-02-23 at 9 58 20 AM](https://github.com/snapsheet/homebrew-core/assets/39897345/9255a516-9bee-465e-9855-8afaa6680119)